packages/eslint-plugin/src/util/astUtils.ts

```diff
@@ -33,7 +33,7 @@ function isNotNonNullAssertionPunctuator(
  */
 function isOptionalOptionalChain(
   node: TSESTree.Node,
-): node is TSESTree.OptionalCallExpression {
+): node is TSESTree.OptionalCallExpression & { optional: true } {
```

Comment:
this type guard was not correct before as `!isOptionalOptionalChain` was not including `OptionalCallExpression` with optional as false